streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -25,24 +25,43 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.Objects;
+
 public class KTableSource<K, V> implements ProcessorSupplier<K, V> {
     private static final Logger LOG = LoggerFactory.getLogger(KTableSource.class);
 
-    public final String storeName;
+    private final String storeName;
+    private String queryableName;
+    private boolean sendOldValues;
 
-    private boolean sendOldValues = false;
+    public KTableSource(final String storeName, final String queryableName) {
```

Comment:
It seems there are two cases here: 
(1) the table is queryable, for this case `storeName == queryableName`
(2) the table is not queryable, for this case `queryableName == null`

Thus, it seems cleaner at this level, to keep `storeName` and `isQueryable` flag to avoid potential bugs that `storeName != queryableName` ?

At outer layers, we can still do it differently. Thoughts?

Reply:
I personally like the distinguishment of queryableName and storeName for both this class and MaterializedInternal :) Let me try to see if I can persuade you:

1. storeName is the name for underlying store no matter if the store does exist or not (i.e. it may not be materialized), as long as the store is materialized the storeName would never be null.

2. queryableName is the name to query the store, it is set once and done. Logically it represents the logic of `isQueryable ? return storeName : return null` in a single place (otherwise there are a bunch of those places where this logic are needed and hence more error prone).

For KTableSource as an example, by the time we create this node we cannot tell yet whether we will materialize it at the end (e.g. if the downstream operators require this ancestor to send old values or materialize, we will then do it). So we need to generate the storeName as always but keep the queryableName as null.

I think the concern that `queryableName != storeName` is valid, for which case I would like to put `generateStoreNameIfNeeded` as part of the MaterializedInternal constructor, and hope that would cleanup a little bit.

Reply:
Ack.